blockchain/state/iterator.go

```diff
@@ -187,14 +186,14 @@ func (it *NodeIterator) retrieve() bool {
 }
 
 // CheckStateConsistency checks the consistency of all state/storage trie of given two state database.
-func CheckStateConsistency(oldDB database.DBManager, newDB database.DBManager, root common.Hash, mapSize int, quit chan struct{}) error {
+func CheckStateConsistency(oldDB Database, newDB Database, root common.Hash, mapSize int, quit chan struct{}) error {
 	// Create and iterate a state trie rooted in a sub-node
-	oldState, err := New(root, NewDatabase(oldDB))
```

Comment:
Not related to this PR, however, why do we have same `NewDatabase` function in both `state` and `statedb` packages?

Reply:
They look really confusing.. we should change their names..

Reply:
Right. Original code has both function. We need to refine.